AK/Vector.h

```diff
@@ -689,8 +689,8 @@ requires(!IsRvalueReference<T>) class Vector {
         MUST(try_resize_and_keep_capacity(new_size));
     }
 
-    using ConstIterator = SimpleIterator<Vector const, T const>;
-    using Iterator = SimpleIterator<Vector, T>;
+    using ConstIterator = SimpleIterator<Vector const, RemoveReference<T> const>;
+    using Iterator = SimpleIterator<Vector, RemoveReference<T>>;
```

Comment:
Please explain what issue this commit is addressing *in this commit's description*.